src/utils.ts

```diff
@@ -413,3 +413,13 @@ export function getProperty(obj: unknown, path: string): unknown {
 
     return current;
 }
+
+/**
+ * Checks if the extension is currently on a virtual file system.
+ */
+export function checkVirtualWorkspace(_actionData: ActionData): ActionValidationResult {
+    if (vscode.workspace.workspaceFolders?.every(f => f.uri.scheme !== 'file')) {
+        return actionValidationFailure('You are currently in a virtual workspace.');
```

Comment:
This should also explain why this fails, instead of just stating that. Otherwise this might sound just like general information.
```suggestion
        return actionValidationFailure('You cannot perform this action in a virtual workspace.');
```

Reply:
Wait why is this outdated

Reply:
weird ways to work git amirite...